src/nvim/screen.c

```diff
@@ -637,12 +637,26 @@ bool win_cursorline_standout(const win_T *wp)
 static DecorationRedrawState decorations;
 bool decorations_active = false;
 
-void decorations_add_luahl_attr(int attr_id,
+void decorations_add_luahl_attr(int attr_id, int prio,
                                 int start_row, int start_col,
                                 int end_row, int end_col)
 {
-  kv_push(decorations.active,
-          ((HlRange){ start_row, start_col, end_row, end_col, attr_id, NULL }));
+  HlRange new = (HlRange){ start_row, start_col, end_row, end_col, prio, attr_id, NULL };
+
+  for (size_t index = 0; index < kv_size(decorations.active); index++) {
```

Comment:
As we're chatting about in gitter, it would be better if we could find a way to not have to loop through all of these every time.

Reply:
Yes, but I really don't know how to fix that though.. The problem is that we want to know if there is actually another match here... Is there any data structure that we can use to mitigate that ?

Reply:
I don't know either, just wanted to post it here in case any one had some ideas haha

Reply:
I am not sure if it is needed anymore now that we are using the match stack thing, in any case, it will be useful in the end, but we really want to be sure that it is a bit more efficient...

Reply:
What is the status here now that we have the match stack stuff. I want to make sure this is up-to-date